core/frontend/src/Viewport.ts

```diff
@@ -737,6 +739,51 @@ export abstract class Viewport implements IDisposable {
     return this.displayStyle.getRealityModelAppearanceOverride(index);
   }
 
+  /** Return the "contextual" reality model index for a transient model ID or -1 if none found
+   * @beta
+   */
+  public getRealityModelIndexFromTransientId(id: Id64String): number {
+    return this.displayStyle.getRealityModelIndexFromTransientId(id);
+  }
+
+  /** Override the planar clip  for a reality model displayed by this viewport.
+  * @param planarClipMask The planar clip mask.
+  * @param modelIdOrIndex The Id of the [[model]] if the attached to the view or the index if it is a context model displayed by this viewport.
+  * @returns true if clip masks are successfully applied.
+  * @beta
+  */
+  public overrideRealityModelPlanarClipMask(modelIdOrIndex: Id64String | number, planarClipMask: PlanarClipMaskSettings): boolean {
+    const changed = this.displayStyle.overrideRealityModelPlanarClipMask(modelIdOrIndex, planarClipMask);
+    if (changed) {
+      this._changeFlags.setDisplayStyle();

```

Comment:
Can you have the display style settings emit an event to which the viewport can respond by invalidating its render plan, so people don't have to call special viewport methods to make their changes visible?

Reply:
Yes... I'll look into this - will save a lot of questions.

Reply:
I realize now that I had already done this and the viewport methods were superflous.   Removed these updated the frontend-dev tools and the comments.   Thanks for pointing this out.